auth/auth.go

```diff
@@ -391,9 +446,14 @@ func (auth *Authenticator) InvalidateChannels(name string, isUser bool, invalSeq
 
 	base.InfofCtx(auth.LogCtx, base.KeyAccess, "Invalidate access of %q", base.UD(name))
 
+	subdocPath := "channel_inval_seq"
+	if scope != base.DefaultScope || collection != base.DefaultCollection {
+		subdocPath = "collection_access." + scope + "." + collection + "." + subdocPath
+	}
+
 	if auth.bucket.IsSupported(sgbucket.DataStoreFeatureSubdocOperations) {
-		err := auth.bucket.SubdocInsert(docID, "channel_inval_seq", 0, invalSeq)
-		if err != nil && err != base.ErrNotFound && err != base.ErrAlreadyExists {
+		err := auth.bucket.SubdocInsert(docID, subdocPath, 0, invalSeq)
+		if err != nil && err != base.ErrNotFound && err != base.ErrAlreadyExists && err != base.ErrPathNotFound {
```

Comment:
How do we get a new error from this? Is this because sometimes we are trying to insert into collection or scopes that don't exist?

I don't feel strongly about this, but in two months I'm not going to remember what this means.

I might consider flattening

```
type ScopeAndCollection {
    Scope string
    Collection string
} 
```

which might make this more readability and flattens out the map. I didn't do a perf analysis on this.

Reply:
Right, subdocPath isn't a root level property when invalidating channels for a collection, and if it's trying to invalidate a collection that doesn't exist, it will get the PathNotFound error.  In this case we don't have anything that needs to be invalidated, so the error can be ignored.